### DISCUSSION TITLE
Make a call to GitHub API from Workflow job

### DISCUSSION THREAD

**Discussion by AlishStrong** [community/community#25226] [Actions]
<p>Hello,</p>
<p>I have a workflow that has 2 jobs. The first job runs angular tests and saves them to tests.log file. The second job just prints the content of the tests.log</p>
<p>What I want however is to post the content of the tests.log file as a general comment on specific Pull Request. So,</p>
<ol>
<li>
<p>Someone makes a Pull Request</p>
</li>
<li>
<p>The workflow gets triggered and it runs the tests.</p>
</li>
<li>
<p>Test results are saved into tests.log file for the next job in the sequence.</p>
</li>
<li>
<p>The next job in the sequence uses the content of tests.log file and posts a comment to the current Pull Request. Content of the comment are test logs from the tests.log file.</p>
</li>
</ol>
<p>I managed with 1-3 steps of my workflow. But for step 4 I don’t understand how to create a comment for a Pull Request; I managed to access and just read logs from tests.logs file. Search results for “GitHub API calls in workflow” and simmilar resulted in how to make API calls to get Workflow or Actions of a project. But I need an opposite thing - I need to call GitHub API inside a workflow</p>

**Comment by hotstone** [comment]
<p>There’s actually an action that calls the Github API: <a href="https://github.com/octokit/request-action" rel="nofollow noopener">https://github.com/octokit/request-action</a></p>
<p>You can use that to create a comment something like:</p>
<pre><code>jobs:
  addCommentToPr:
    runs-on: ubuntu-latest
    steps:
      - uses: octokit/request-action@v2.x
        id: add_pr_comment
        with:
          route: POST /repos/:repository/issues/:issue_number/comments
          repository: ${{ github.repository }}
          issue_number: ${{ github.event.number }}
          body: A test comment
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>It seems weird that I had to pass the pull request number as the issue number, and it may not be entirely correct, but it certainly worked when I tested it.  Here’s the documentation on the API about comments on PRs (you’ve already read this I’m sure): <a href="https://developer.github.com/v3/pulls/comments/" rel="nofollow noopener">https://developer.github.com/v3/pulls/comments/</a></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/alishstrong">@alishstrong</a> , </p>
<p>There is <a href="https://github.com/marketplace/actions/comment-on-pr" rel="nofollow noopener">an action</a> to add comment on PR, the comment content is from a file .  If use GITHUB_TOKEN , the comment is added by github-actions bot . </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3c36196a15e9f8722a2cc64d6ec415ed45795765.png" data-download-href="/uploads/short-url/8AEyqULa07WT853mhZW3zXDHJE9.png?dl=1" title="PR comments.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089691-cc76f04b-cf0b-47c9-a9a8-4439bdeb843b.png" alt="PR comments.png" data-base62-sha1="8AEyqULa07WT853mhZW3zXDHJE9" width="473" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3c36196a15e9f8722a2cc64d6ec415ed45795765_2_473x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/3c36196a15e9f8722a2cc64d6ec415ed45795765_2_709x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/3c36196a15e9f8722a2cc64d6ec415ed45795765.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3c36196a15e9f8722a2cc64d6ec415ed45795765_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">PR comments.png</span><span class="informations">777×820 36.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The file path need to be a relative path against  .github/workflows . For example, your logs file is located in .github/workflows , then just specify the logs file name in finename variable. </p>
<p>If the logs file is under the root path of your repo, you need to use …/…/tests.log . </p>

**Comment by miigotu** [comment]
<p>This is the more correct solution. The API action is very versatile and exactly what I was looking for!</p>
